pennylane_ionq/device.py

```diff
@@ -235,34 +380,164 @@ def _submit_job(self):
         # state (as a base-10 integer string) to the probability
         # as a floating point value between 0 and 1.
         # e.g., {"0": 0.413, "9": 0.111, "17": 0.476}
-        self.histogram = job.data.value
+        some_inner_value = next(iter(job.data.value.values()))
+        if isinstance(some_inner_value, dict):
+            self.histograms = []
+            for key in job.data.value.keys():
+                self.histograms.append(job.data.value[key])
+        else:
+            self.histograms = []
+            self.histograms.append(job.data.value)
+            self.histogram = job.data.value
 
     @property
     def prob(self):
         """None or array[float]: Array of computational basis state probabilities. If
         no job has been submitted, returns ``None``.
         """
+        if self._current_circuit_index is not None:
+            self.histogram = self.histograms[self._current_circuit_index]
+
         if self.histogram is None:
             return None
 

```

Comment:
I removed self.histogram. However, self._current_circuit_index remains defaulted to None instead 0. Here is the reason for that: when the number of circuits is larger than one and the user does not self._current_circuit_index an exception is raised notifying the user he must set a _current_circuit_index. If I would set _current_circuit_index 0  by default, a user which would use functions like dev.probability() or dev.sample() will always retrieve results for the first circuit which is incorrect and it would be very difficult for the user to notice he made a mistake. However 99% of the user will not use functions that are internal to device implementation like dev.probability() or dev.sample() because they would just submit a batch of circuits and then retrieve a batch of results and the whole _current_circuit_index issue will not be visible to them.